hazelcast/src/main/java/com/hazelcast/concurrent/lock/LockServiceImpl.java

```diff
@@ -258,10 +258,20 @@ public void commitMigration(PartitionMigrationEvent event) {
             clearLockStoresHavingLesserBackupCountThan(event.getPartitionId(), event.getNewReplicaIndex());
         } else {
             int partitionId = event.getPartitionId();
+            // Local locks are local to the partition and replicaIndex where they have been acquired.
+            // That is the reason they are removed on any partition event on the destination.
+            removeLocalLocks(partitionId);
```

Comment:
Shouldn't `removeLocalLocks()` be called on `SOURCE`? They won't be replicated to destination, right?